BricksAndHearts.UnitTests/ServiceTests/Property/PropertyServiceTests.cs

```diff
@@ -1,31 +1,255 @@
+    {
+        // Arrange
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+        context.Properties.Where(u => u.Id == 1).ToList()[0].AcceptsBenefits = true;
+        var propertiesBeforeCount = context.Properties.Count();
+        var updateModel = new PropertyViewModel
+        {
+            Address = new PropertyAddress
+            {
+                AddressLine1 = "UpdateProperty_UpdatesProperty1",
+                County = "Cambridgeshire",
+                TownOrCity = "Cambridge",
+                Postcode = "CB1 1DX"
+            },
+            AcceptsBenefits = false
+        };
+
+        // Act
+        service.UpdateProperty(1, updateModel, false);
+        context.ChangeTracker.Clear();
+
+        // Assert
+        var property = context.Properties.Where(u => u.Id == 1).ToList()[0];
+        property.AddressLine1.Should().Be("UpdateProperty_UpdatesProperty1");
+        property.IsIncomplete = false;
+        property.AcceptsBenefits = false;
+        context.Properties.Count().Should().Be(propertiesBeforeCount);
+    }
+
+    #endregion
+
+    #region DeleteProperty
+
+    [Fact]
+    public async void DeleteProperty_DeletesProperty()
+    {
+        // Arrange
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+        var propertiesBeforeCount = context.Properties.Count();
+        var deleteModel = context.Properties.Where(u => u.Id == 1).ToList()[0];
+
+        // Act
+        service.DeleteProperty(deleteModel);
+        context.ChangeTracker.Clear();
+
+        // Assert
+        context.Properties.Count(u => u.Id == 1).Should().Be(0);
+        context.Properties.Count().Should().Be(propertiesBeforeCount - 1);
+    }
+
+    #endregion
+
+    #region CountProperties()
+
+    [Fact]
+    public async void CountProperties_ReturnsCorrectCountsOfAll()
+    {
+        // Arrange
+        await using var context = Fixture.CreateReadContext();
+        var service = new PropertyService(context);
+
+        // Act
+        var propertyCount = service.CountProperties();
+        context.ChangeTracker.Clear();
+
+        // Assert
+        propertyCount.RegisteredProperties.Should().Be(context.Properties.Count());
+    }
+
+    #endregion
+
+    #region IsUserAdminOrCorrectLandlord
+
+    [Theory]
+    [InlineData(1)]
+    [InlineData(2)]
+    public async void IsUserAdminOrCorrectLandlord_UsedByAdmin_ReturnsTrueEvenIfPropertyIsntTheirs(int OwnerId)
+    {
+        // Arrange
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+
+
+        var adminUser = CreateAdminUser();
+
+
+        context.Properties.First(u => u.Id == 1).LandlordId = OwnerId;
+
+        // Act
+        var allowed = service.IsUserAdminOrCorrectLandlord(adminUser, 1);
+        context.ChangeTracker.Clear();

```

Comment:
This test is a bit confusing. I don't think the `[InlineData]` helps here. We just need to make sure the landlord Id is different from the admin's user id. We can do that by setting the admin user id like you do below, so let's do that.

The `context.ChangeTracker.Clear();` call here also doesn't do anything since our asserts don't use the database.